docs/conf.py

```diff
@@ -102,14 +102,21 @@
 # documentation created by autosummary uses a template file (in autosummary in the templates path),
 # which likely overrides the autodoc defaults.
 
+# These options impact when using `.. autoclass::` manually.
+# They do not impact the `.. autosummary::` templates.
+autodoc_default_options = {
+    "show-inheritance": True,
+}
+
 # Move type hints from signatures to the parameter descriptions (except in overload cases, where
 # that's not possible).
 autodoc_typehints = "description"
-# Only add type hints from signature to description body if the parameter has documentation.  The
-# return type is always added to the description (if in the signature).
-autodoc_typehints_description_target = "documented_params"
-
 autoclass_content = "both"
+# Some type hints are too long to be understandable. So, we set up aliases to be used instead.
+autodoc_type_aliases = {
+    "EstimatorPubLike": "EstimatorPubLike",
+    "SamplerPubLike": "SamplerPubLike",
+}
```

Comment:
Feel free to disagree on this. The Runtime devs thought in https://github.com/Qiskit/qiskit-ibm-runtime/issues/1877 that the full type is way too long to actually be understandable. I agree with them, and I think it's better to explain pubs in guides like https://docs.quantum.ibm.com/guides/primitive-input-output. But there's an argument for having the full type.

Reply:
I agree with the aliases, the full type is not really helpful.